lib/Transforms/Scalar/ScalarReplAggregatesHLSL.cpp

```diff
@@ -1828,13 +1828,13 @@ bool SROAGlobalAndAllocas(HLModule &HLM, bool bHasDbgInfo) {
       bool SROAed = false;
       if (GlobalVariable *NewEltGV = dyn_cast_or_null<GlobalVariable>(
               TranslatePtrIfUsedByLoweredFn(GV, typeSys))) {
+        GVDbgOffset dbgOffset = GVDbgOffsetMap[GV];
```

Comment:
I think there should be an additional if (GV != NewEltGV) guarding the code here, since nothing needs to be done, other than avoiding SROA in the case they are equal.  But the current code is equivalent, if not efficient in that case.